torch/nn/modules/module.py

```diff
@@ -1563,6 +1594,18 @@ def load(module, prefix=''):
                 if child is not None:
                     load(child, prefix + name + '.')
 
+            # Note that the hook can modify missing_keys and unexpected_keys.
+            incompatible_keys = _IncompatibleKeys(missing_keys, unexpected_keys)
+            for hook in module._load_state_dict_post_hooks.values():
+                out = hook(module, incompatible_keys)
+                # load_state_dict post hooks are not expected to return new
+                # values, if incompatible_keys can be modified, it should be done
+                # inplace.
+                assert out is None, (
+                    "Expected hook registered with ``register_load_state_dict_post_hook``"
+                    " to return ``None``."
```

Comment:
I would just move the whole comment above in this error message. No need for duplication and it is good to have a hint at what to do when you get the error.